dev-client/src/screens/HomeScreen/components/MapSearch.tsx

```diff
@@ -211,14 +214,22 @@ export default function MapSearch({zoomTo, zoomToUser, toggleMapLayer}: Props) {
             padding={2}
             onPress={toggleMapLayer}
           />
-          <IconButton
-            name="my-location"
-            _icon={{color: 'action.active'}}
-            bgColor="white"
-            padding={2}
-            borderRadius={15}
-            onPress={zoomToUser}
-          />
+          <PermissionsRequestModal
+            title={t('permissions.location_title')}
+            body={t('permissions.location_body')}
+            usePermissions={useForegroundPermissions}
+            permissionedAction={zoomToUser}>
+            {onRequest => (
+              <IconButton
+                name="my-location"
+                _icon={{color: 'action.active'}}
+                bgColor="white"
+                padding={2}
+                borderRadius={15}
+                onPress={onRequest}
+              />
+            )}
+          </PermissionsRequestModal>
```

Comment:
Sorta-suggestion: I've seen a few places in our codebase this model where the component hierarchy has a modal that is opened by pressing a button, and the button defined within its onRequest. It reads a little convoluted to me, because I tend to expect to see the button in the hierarchy because that's what's visible.

If you'd prefer not to refactor it that way here, perhaps this can be a future code standard discussion -- but wanted to note it!

Reply:
discussed on call, doing a rename for now but considering changing structurally in future